core/src/commonMain/kotlin/Error.kt

```diff
@@ -0,0 +1,3 @@
+package com.juul.kable
+
+public expect class Error
```

Comment:
Would this make more sense as an `interface`? I can imagine a source set wanting more than one error type.

Reply:
I really need to read up more on multiplatform and exceptions. This is here to represent the following:

```
// Android
public actual typealias Error = Throwable

// Javascript
public actual typealias Error = Throwable

// Apple
public actual typealias Error = NSError
```

So, mostly to accommodate when a `data class` is going to carry an error state (i.e. `NSError` on Apple platform).
Although perhaps multiplatform will automatically convert `Throwable` to `NSError` on the Apple platform.

I'll look into it some more.

Reply:
There are really just two pieces of info in an `NSError` that we would want to surface - an error code (integer which maps to a `#define` in the C header files so we can find the exact underlying system fault) and a string reason message. There are several accessors for obtaining the string but probably the `localizedDescription` method is the best choice.

If you wanted to distill it down to a single "reason" string or something, you could just build a new string that includes both the code and the "description" and avoid having to carry that extra field.

So, perhaps use any public facing API shape you want, so long as it can surface said string?

Reply:
Thanks @davidtaylor-juul, that's exactly the info I needed.

I'll probably end up with a custom exception like:

```kotlin
class NsErrorException(
    val localizedDescription: String,
    val errorCode: Int
) : IllegalStateException("$errorCode: $localizedDescription")
```

So that the exception message carries the needed info when viewed in logs and the specific values can be accessed if the exception type is known at a call site.